File: E-Commerce-Site/E-Commerce-Site/Controllers/UsersController.cs
namespace E_Commerce_Site.Controllers;

using Microsoft.AspNetCore.Mvc;
using Users;
using Users.Models;

[ApiController]
public class UsersController : ControllerBase
{
    
    private readonly IUsersService _usersService;
    
    public UsersController(IUsersService usersService)
    {
        _usersService = usersService;
    }

    [HttpPost("api/addUser")]
    public async Task AddUser([FromBody] User definition)
    {
        await _usersService.AddUser(definition);
    }
    
    [HttpGet("api/availableUsername/{username}")]
    public async Task<IActionResult> GetAvailableUser([FromRoute] string username)
    {
        bool available = await _usersService.GetAvailableUser(username);
        return Ok(available);
    }
    
    [HttpPost("api/login")]
    public async Task<IActionResult> Login([FromBody] User definition)
    {
        bool successfulLLogin = await _usersService.Login(definition);
        return Ok(successfulLLogin);
    }
}

File: Python_BackEnd/Controllers/UsersController.py
from fastapi import Request, APIRouter
import Users.UsersService as UsersService
from Users.Models.Users import User
import uuid

router = APIRouter()


@router.get("/api/availableUsername/{username}")
async def get_available_user(username: str) -> bool:
    details = await UsersService.get_available_user(username)
    return details


@router.post("/api/addUser")
async def add_user(request: Request) -> None:
    data = await request.json()
    user_instance = User.from_dict(data)
    await UsersService.add_user(user_instance)


@router.post("/api/login")
async def login(request: Request):
    data = await request.json()
    user_instance = User.from_dict(data)
    return await UsersService.login(user_instance)
